kong/clustering/services/sync/init.lua

```diff
@@ -0,0 +1,69 @@
+local _M = {}
+local _MT = { __index = _M, }
+
+
+local events = require("kong.clustering.events")
+local strategy = require("kong.clustering.services.sync.strategies.postgres")
+local rpc = require("kong.clustering.services.sync.rpc")
+
+
+-- TODO: what is the proper value?
+local FIRST_SYNC_DELAY = 0.5  -- seconds
+local EACH_SYNC_DELAY  = 30   -- seconds
+
+
+function _M.new(db, is_cp)
+  local strategy = strategy.new(db)
+
+  local self = {
+    db = db,
+    strategy = strategy,
+    rpc = rpc.new(strategy),
+    is_cp = is_cp,
+  }
+
+  -- only cp needs hooks
+  if is_cp then
+    self.hooks = require("kong.clustering.services.sync.hooks").new(strategy)
```

Comment:
style suggestion: (not a blocker, we could resolve it later)

We'd better call `hooks.new()` and define a `local hooks` library variable like `local hooks = require("kong.clustering.services.sync.hooks")`, which aligns with other required libraries like `strategy.new()` and `rpc.new()`

Reply:
`hooks` is only needed in CP, if it is a module level variable I think that it will have some extra memory overload.

Reply:
We could find the similar usage in `clustering/init.lua`

Reply:
That sounds reasonable.